### ISSUE TITLE
Infracost shows no output for ASG module with LaunchConfiguration

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#1180]
This can be reproduced by running `infracost breakdown` against the following Terraform code:

```hcl
provider "aws" {
  region = "us-east-1"
}

module  "asg" {
  source = "terraform-aws-modules/autoscaling/aws"
  version = "~> 4"
  name = "asg"
  use_lc = true
  create_lc = true
  lc_name = "lc"
  image_id = "ami-0ff8a91507f77f867"
  instance_type = "t3.micro"
  min_size = 0
  max_size = 2
  desired_capacity = 1
}
```

Expected output (similar to):
```
 Name                                                    Monthly Qty  Unit     Monthly Cost

 aws_autoscaling_group.asg
 └─ aws_launch_configuration.lc
    ├─ Instance usage (Linux/UNIX, on-demand, t3.micro)          730  hours           $7.59
    ├─ EC2 detailed monitoring                                     7  metrics         $2.10
    └─ root_block_device
       └─ Storage (general purpose SSD, gp2)                       8  GB              $0.80

 OVERALL TOTAL                                                                       $10.49
```

Actual output:
```
Project: .

 Name  Monthly Qty  Unit  Monthly Cost

 OVERALL TOTAL                   $0.00
```

**Comment by aliscott** [comment]
The issue is the plan file shows no relation between the ASG and the launch configuration address, so Infracost isn't creating the reference. The references section of the ASG in the plan JSON shows:

```
  "launch_configuration": {
    "references": [
      "var.use_lc",
      "local.launch_configuration"
    ]
  },
```

But since the plan file doesn't contain the evaluated value of `local` variables then we have no way of knowing which resource `local.launch_configuration` refers to.

I've added a fix for this module by adding a special mapping, which we've done in the past for other modules here: https://github.com/infracost/infracost/pull/1185.

**Comment by aliscott** [comment]
This has been fixed in https://github.com/infracost/infracost/pull/1185 and will be released in v0.9.15 which we will release at nearer the end of this week/beginning of next week.

**Comment by aliscott** [comment]
This has now been released in v0.9.15.
